service/src/main/java/org/jboss/sbomer/service/feature/sbom/k8s/reconciler/GenerationRequestReconciler.java

```diff
@@ -147,24 +147,27 @@ public GenerationRequestReconciler() {
     private UpdateControl<GenerationRequest> reconcileNew(
             GenerationRequest generationRequest,
             Set<TaskRun> secondaryResources) {
-        TaskRun initTaskRun = initTaskRun(secondaryResources);
 
-        if (initTaskRun == null) {
-            return UpdateControl.noUpdate();
-        }
-
-        if (isFinished(initTaskRun)) {
-            if (isSuccessful(initTaskRun)) {
-                generationRequest.setStatus(SbomGenerationStatus.INITIALIZED);
-                setConfig(generationRequest, initTaskRun);
-            } else {
-                generationRequest.setStatus(SbomGenerationStatus.FAILED);
-            }
-        } else {
-            generationRequest.setStatus(SbomGenerationStatus.INITIALIZING);
-        }
-
-        return UpdateControl.updateResource(generationRequest);
+        return initTaskRun(secondaryResources)
+            .map(initTaskRun -> {
+                    if (isFinished(initTaskRun)) {
+                        isSuccessful(initTaskRun)
+                            .ifPresentOrElse(isSuccessful -> {
+                                    if (isSuccessful) {
+                                        generationRequest.setStatus(SbomGenerationStatus.INITIALIZED);
+                                        setConfig(generationRequest, initTaskRun);
+                                    } else {
+                                        generationRequest.setStatus(SbomGenerationStatus.FAILED);
+                                    }
+                                },
+                                () -> new IllegalStateException("Finished task should return non-empty value.")
+                            );
+                    } else {
+                        generationRequest.setStatus(SbomGenerationStatus.INITIALIZING);
+                    }
+                    return UpdateControl.updateResource(generationRequest);
+                })
+            .orElseGet(() -> UpdateControl.noUpdate());
```

Comment:
I have mixed feelings about this. It became much more hard to read IMHO. @vibe13 WDYT?

Reply:
Mmmmm wow yes I have a very hard time reading the code as well.